app/components/dashboard-interactive-filters.tsx

```diff
@@ -213,3 +238,180 @@ function stepFromTimeUnit(timeUnit: TimeUnit | undefined) {
+        .map((cube) => cube.iri)
+    )
+  );
+
+  if (cubeIris.length > 1) {
+    console.error(
+      `Data filter ${componentIri} is used in multiple cubes: ${cubeIris.join(", ")}`
+    );
+  }
+
+  const cubeIri = cubeIris[0];
+
+  const [{ data }] = useDataCubesComponentsQuery({
+    variables: {
+      sourceType: state.dataSource.type,
+      sourceUrl: state.dataSource.url,
+      locale,
+      cubeFilters: [
+        { iri: cubeIri, componentIris: [componentIri], loadValues: true },
+      ],
+    },
+    keepPreviousData: true,
+  });
+
+  const dimension = data?.dataCubesComponents.dimensions[0];
+
+  const [value, setValue] = useState<string>();
+  const handleChange = useEvent(
+    (
+      e:
+        | SelectChangeEvent<unknown>
+        | ChangeEvent<HTMLSelectElement>
+        | { target: { value: string } }
+    ) => {
+      const newValue = e.target.value as string;
+      setValue(newValue);
+
+      for (const [chartKey, [_getState, _useStore, store]] of Object.entries(
+        dashboardInteractiveFilters.stores
+      )) {
+        if (chartConfigs.map((config) => config.key).includes(chartKey)) {
+          store.setState({
+            dataFilters: {
+              ...store.getState().dataFilters,
+              [componentIri]: {
+                type: "single",
+                value: newValue,
+              },
+            },
+          });
+        }
+      }
+    }
+  );
+
+  // Store the state of the stores to restore it when the component is unmounted
+  const storesRef = useRef<{ [chartKey: string]: InteractiveFiltersState }>();
+  useEffect(() => {
+    storesRef.current = Object.fromEntries(
+      Object.entries(dashboardInteractiveFilters.stores).map(
+        ([key, [_getState, _useStore, store]]) => {
+          return [key, store.getState()];
+        }
+      )
+    );
+
+    if (dimension?.values.length) {
+      handleChange({
+        target: { value: dimension.values[0].value as string },
+      } as ChangeEvent<HTMLSelectElement>);
+    }
+
+    const storesRefCurrent = storesRef.current;
+
+    return () => {
+      for (const [chartKey, [_getState, _useStore, store]] of Object.entries(
+        dashboardInteractiveFilters.stores
+      )) {
+        if (chartConfigs.map((config) => config.key).includes(chartKey)) {
+          const dataFilters = store.getState().dataFilters;
+          const refStore = storesRefCurrent[chartKey];
+          if (refStore) {
+            const refDataFilters = refStore.dataFilters;
+            if (refDataFilters[componentIri]) {
+              dataFilters[componentIri] = refDataFilters[componentIri];
+              store.setState({ dataFilters });
+            } else {
+              delete dataFilters[componentIri];
+              store.setState({ dataFilters });
+            }
+          }
+        }
+      }
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [dimension]);
```

Comment:
Not sure about this.

- I do not think we should so tightly bind the component display and the store save/restore functionality
- I am not sure the component should be this knowledgeable about the store.
- Could the store actually provide functions that save/store this ?

I think I would be more happy with something like this

```

const { saveDataFilters, restoreDataFilters } = useDashboardInteractiveFilters()
useEffect(() => {
  const snapshot = saveDataFiltersSnapshot(componentIri)
  return () => {
      restoreDataFiltersSnapshot(componentIri)
  })

}, [dimensionIri])
```

Then I am still thinking that this should not be tied to UI but more to chartConfigs state. Actually it seems to me that useDashboardInteractiveFilters could maybe completely hide this (react on changes to the config to see if a data filters is toggled off/on and save/restore accordingly) ?

What do you think ?

Reply:
For a bit of context, here I also wanted to be consistent about how we handle this with dashboard time slider, where we also keep the "dashboard time filter state" inside the time slider, and just update the individual chart stores. Maybe the logic here got a bit too complex with saving / restoring things 😅 

I agree it would make it a bit easier to handle things by moving the state up for both areas, time filter and categorical filters to the dashboard state 👍